raven/utilities/forecasting.py

```diff
@@ -173,3 +182,98 @@ def perform_climatology_esp(model_name, forecast_date, forecast_duration, **kwds
+    '''
+   
+    # Get the file locations and filenames as a function of the climate model and date
+    if climate_model is 'GEPS':
+        
+        # Forecasts will always be recent so start with today!
+        date_now=dt.datetime.now()
+
+        # Find the most recent data on the server by looping from today and go
+        #backwards 5 days. If no data is present, not much value in forecasting.
+        filein=None
+        for i in range(0,7):
+            
+            # Write the string
+            date=date_now-dt.timedelta(days=i)
+            datefcst=dt.datetime.strftime(date,"%Y%m%d")
+            
+            # try to open the files. If it works, great, if no
+            try:
+                filein='https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/testdata/geps_forecast/CMC_geps-raw_latlon0p5x0p5_' + datefcst + '00_allP_allmbrs.nc'
+                ds=xr.open_dataset(filein)
+                # Checking that these exist. IF the files are still processing, possible that one or both are not available!
+                assert 'pr' in ds
+                assert 'tas' in ds
+                break
+            except:
+                filein=None
+        
+        # If we have not found any GEPS data, raise error.
+        if filein is None:
+            raise RuntimeError('Unable to locate valid forecast data up to 5 days prior to today.')
+
+
+        
+        # Here we also extract the times at 6-hour intervals as Raven must have 
+        #constant timesteps.
+        times=[]
+        date = dt.datetime(date.year, date.month, date.day)
+        for single_time in (date + dt.timedelta(hours=n) for n in range(0,384,6)):
+            times.append(single_time)
+
+    elif climate_model is 'GDPS':
+        filein='https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/caspar/daily/GDPS_' + dt.datetime.strftime(date,'%Y%m%d') + '.nc'
+
+    elif climate_model is 'RDPS':
+        filein='https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/caspar/daily/REPS_' + dt.datetime.strftime(date,'%Y%m%d') + '.nc'
+
+    elif climate_model is 'REPS':
+        filein='https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/caspar/daily/RDPS_' + dt.datetime.strftime(date,'%Y%m%d') + '.nc'
+
+    # Extract the shapefile from the zipped package
+    tmp = Path(tempfile.mkdtemp())
+    ZipFile(region,'r').extractall(tmp)
+    shp = list(tmp.glob("*.shp"))[0]
+    vector = fiona.open(shp, "r")
+    shdf = [vector.next()["geometry"]]
+
+    # extract the bounding box to subset the entire forecast grid to something
+    # more manageable
+    lon_min=vector.bounds[0]
+    lon_max=vector.bounds[2]
+    lat_min=vector.bounds[1]
+    lat_max=vector.bounds[3]
+    
+    # Subset the data to the desired location (bounding box) and times
+    ds=subset.subset_bbox(ds, lon_bnds=[lon_min, lon_max], lat_bnds=[lat_min, lat_max]).sel(time=times)
```

Comment:
Another thing with the control flow I'm not sure I understand: if `climate_model` is not `"GEPS"`, `times` is not defined, so this will crash.

Reply:
yes, it's because we have no files for these models yet. Times will be added when we get that info. For now, only GEPS will work though.

Reply:
also, I should mention that the only 4 options that will ever be available are GEPS, REPS, GDPS and RDPS. so perhaps we should have a sanitizer or checker of some sort to make sure that this is the case. (but only GEPS for now).